proto/minder/v1/minder.proto

```diff
@@ -1282,17 +1282,22 @@ message RuleType {
                     string action = 2;
                     // the content of the file
                     string content = 4;
-                    // the GIT mode of the file. Not UNIX mode! String because the GH API also uses strings
-                    // the usual modes are: 100644 for regular files, 100755 for executable files and
-                    // 040000 for submodules (which we don't use but now you know the meaning of the 1 in 100644)
+                    // The UNIX file permissions to set on the file
```

Comment:
actually, I was stupid and git doesn't support the full set of UNIX permissions, but a subset of them plus some of its own (see e.g. https://github.com/go-git/go-git/blob/32e0172851c35ae2fac495069c923330040903d2/plumbing/filemode/filemode.go#L10)

So I reverted to using the git mode back and I actually used go-git's utility functions to convert the git modes into unix mode with error handling.

Since we're squashing PR patches, I'll be pushing the changes as individual patches for easier review.